pkg/kubelet/kubelet_node_status.go

```diff
@@ -129,6 +129,24 @@ func (kl *Kubelet) reconcileHugePageResource(initialNode, existingNode *v1.Node)
 	requiresUpdate := false
 	supportedHugePageResources := sets.String{}
 
+	if existingNode.Status.Capacity == nil {
```

Comment:
Could consider factoring out this code (and the block below) which would allow for some extra unit-tests.

Reply:
Thank you for your review @NicolasT . Is there any more detailed reconstruction plan?

Reply:
Not sure I understand the question? You basically have two (+-) 12-line code blocks that are roughly the same, parametrized over two `Node` objects, 'returning' (so to say) some `bool`. Sounds like a good opportunity to turn that functionality into a function which then you call twice, and for which some unit-test(s) can be written.

Reply:
Sounds good!

Reply:
Done!